hypothesis-python/src/hypothesis/database.py

```diff
@@ -675,6 +677,43 @@ def delete(self, key: bytes, value: bytes) -> None:
         raise RuntimeError(self._read_only_message)
 
 
+class BackgroundWriteDatabase(ExampleDatabase):
+    """A wrapper which defers writes on the given database to a background thread.
+    Reads are still executed synchronously on the calling thread.
+    """
+
+    def __init__(self, db: ExampleDatabase) -> None:
+        self._db = db
+        self._queue: Queue[tuple[str, tuple[bytes, ...]]] = Queue()
+        self._thread = Thread(target=self._worker, daemon=True)
+        self._thread.start()
+
```

Comment:
I'm tempted to add a `weakref.finalize(self, self.join)` here, but I'm not sure if it's good practice to have a finalizer that can take an effectively unbounded amount of time. (I'm going to guess no 😄).

Reply:
Maybe in a `__del__` method, and then give it a pretty short timeout?  0.1s should be plenty for most purposes but still well short of unbounded.